services/src/main/java/org/fao/geonet/api/registries/DirectoryUtils.java

```diff
@@ -430,4 +431,31 @@ private static String search(ServiceContext context, Map<String, String> searchP
         }
         return null;
     }
+
+    public static Element removeUnsedLangsEntry(Element entry, final String[] langs) throws JDOMException {
+        List<Element> multilangElement = (List<Element>)Xml.selectNodes(entry, "*//node()[@locale]");
+
+        List<String> twoCharLangs = new ArrayList<String>();
+        for(String l : langs) {
+            twoCharLangs.add("#" + XslUtil.twoCharLangCode(l).toUpperCase());
+        }
+
+        for(Element el : multilangElement) {
+            if(!twoCharLangs.contains(el.getAttribute("locale").getValue())) {
+                Element parent = (Element)el.getParent();
+                parent.detach();
+            }
+        }
+        return entry;
+    }
+
+    public static Element removeMultilingualEntry(Element entry) throws JDOMException {
```

Comment:
I would move this to the SchemaPlugin like https://github.com/geonetwork/core-geonetwork/blob/develop/schemas/iso19139/src/main/java/org/fao/geonet/schema/iso19139/ISO19139SchemaPlugin.java#L203